docs/Configuring-Orion/TLS.md

```diff
@@ -3,15 +3,162 @@ description: TLS
+clientport = 8888
+
+othernodes = ["<YOUR-COMMON-NAME>:8080"]
+
+workdir = "<PATH-TO>/orion/bin"
+
+publickeys = ["orion.pub"]
+privatekeys = ["orion.key"]
+
+tls = "strict"
+
+tlsservercert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsserverchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsserverkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclientcert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsclientchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsclientkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclienttrust = "ca"
+tlsservertrust = "ca"
+
+nodenetworkinterface = "0.0.0.0"
+clientnetworkinterface = "0.0.0.0"
+```
+
+### Generating certificates when the public IP address is the Common Name (CN) 
+
+If you want to use a public IP as the CN, complete the following steps:
+
+#### Updating the `openssl.cfn` file
+
+1. Find the `openssl.cfn` file, and create a copy of it.
+1. In your copy of the `openssl.cfn` file, find the `[req]` section, and add the following lines:
+
+  ```
+  req_extensions = v3_req
+  
+  [ v3_req ]
+  basicConstraints = CA:FALSE
+  keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+  subjectAltName = @alt_names
+
+  [alt_names]
+  DNS.1 = <DNS-PUBLIC-RECORD>
+  DNS.2 = <DNS-PRIVATE-RECORD>
+  IP.1 = <PUBLIC-IP-ADDRESS>
+  IP.2 = <PRIVATE-IP-ADDRESS>
+  ```
+
+1. For each DNS that you want to use as an alternate name, specify a DNS._n_ entry.
+1. For each IP address that you want as an alternate IP address, specify an IP._n_ entry.
+
+#### Generating a new CSR for a node
+
+1. Run the following command. Ensure you substitute your values for all variables.
+
+  `openssl req -new -key orion_cer.key -out orion_cer.csr -config <PATH-TO>/openssl.cnf`
+
+1. Test whether the certificate was generated with the expected subject alternative names by running the following command:
+
+  `openssl req -text -noout -in orion_cer.csr`
+
+The following example shows the type of output this command generates:
+
+```
+[...]
+Requested Extensions:
+            X509v3 Subject Alternative Name:            
+DNS:<DNS-PUBLIC-RECORD>,
+DNS:<DNS-PRIVATE-RECORD>, 
+IP Address:<PUBLIC-IP-ADDRESS>, 
+IP Address:<PRIVATE-IP-ADDRESS>
+[...]
+```
+
+#### Generating a new certificate
+
+1. Run the following command. Ensure you substitute your values for all variables.
+
+  `openssl x509 -req -in orion_cer.csr -CA orion_ca.pem -CAkey orion_ca.key -CAcreateserial -out orion_cer.pem -days 500 -sha256 -extfile <PATH-TO>/openssl.cnf -extensions v3_req`
+
+1. Test whether the generated certificate contains the subject alternative names by running the following command:
+
+  `openssl x509 -in orion_cer.pem -text -noout`
+
+The following example shows the type of output this command generates:
+
+```
+[...]
+X509v3 extensions:
+          X509v3 Subject Alternative Name:
+DNS:<DNS-PUBLIC-RECORD>,
+DNS:<DNS-PRIVATE-RECORD>, 
+IP Address:<PUBLIC-IP-ADDRESS>, 
+IP Address:<PRIVATE-IP-ADDRESS>
+[...]
+```
 
 ## Configuring TLS 
```

Comment:
I don't think this title makes sense anymore now we've added all the new content above it.  The content above it is all about configuring TLS as well.  Maybe something like 'TLS Properties in Orion Configuration File'